rewards/utils/rewards_utils.py

```diff
@@ -78,6 +79,32 @@ def distribute_rewards_to_snapshot(
     return combine_rewards([rewards] + custom_rewards_list, 0)
 
 
+def distribute_rewards_from_total_snapshot(
+        amount: int, snapshot: Snapshot, token: str,
+        custom_rewards: Optional[Dict[str, Callable]] = None,
+):
+    if not custom_rewards:
+        custom_rewards = {}
+    rewards = RewardsList()
+    custom_rewards_list = []
+    total = snapshot.total_balance()
+    # TODO: Think about refactoring this and splitting it into two separate funcs:
+    # TODO: one for normal rewards another for custom rewards
+    for addr, balance in snapshot:
+        if addr in custom_rewards:
+            custom_rewards_calc = custom_rewards[addr]
+            console.log(token, amount, snapshot.token)
+            custom_rewards_list.append(
+                custom_rewards_calc(amount, token, snapshot.token)
```

Comment:
If you check the unclaimed rewards handlers, we have an extra parameter here called block that needs to be passed in. Maybe we should instead pass a dict to this function so that we don't end up with a bunch of unused parameters in our handlers.